### ISSUE TITLE
About the caculation of LEConv

### ISSUE THREAD

**Issue by lilybud** [malllabiisc/ASAP#3]
hi, this is an instresting and excellent paper! After reading the code, i have a problem about the partion of LEConv.py. 

`def forward(self, x, edge_index, edge_weight=None, size=None):
        num_nodes = x.shape[0]
        h = torch.matmul(x, self.weight)

        if edge_weight is None:
            edge_weight = torch.ones((edge_index.size(1), ),
                                     dtype=x.dtype,
                                     device=edge_index.device)
        edge_index, edge_weight = remove_self_loops(edge_index=edge_index, edge_attr=edge_weight)
        deg = scatter_add(edge_weight, edge_index[0], dim=0, dim_size=num_nodes)  # + 1e-10

        h_j = edge_weight.view(-1, 1) * h[edge_index[1]]
        aggr_out = scatter_add(h_j, edge_index[0], dim=0, dim_size=num_nodes)
        out = ( deg.view(-1, 1) * self.lin1(x) + aggr_out) + self.lin2(x)       # ----------the question is here----------# 
        edge_index, edge_weight = add_self_loops(edge_index=edge_index, edge_weight=edge_weight, num_nodes=num_nodes)
        return out`

According to the paper's description,  i think the caculation of out may as follows：
`out = ( deg.view(-1, 1) * self.lin1(x) - aggr_out) + self.lin2(x)   `

i may be wrong and I want to know where my understanding has gone wrong. Can you explain that? 
Thank you~

**Comment by ekagra-ranjan** [comment]
Hi,

Let `self.weight` in `h = torch.matmul(x, self.weight)` be represented as `W*`. Since `self.weight` is a learnable quantity, it can converge to `-W*` if needed and hence `aggr_out` will also become -ve. Hope this helps.
